pyscal/gasoil.py

```diff
@@ -238,10 +238,9 @@ def add_fromtable(
             dframe = dframe.reset_index()
 
         if sgcolname not in dframe:
-            logger.critical(
-                "%s not found in dataframe, can't read table data", sgcolname
+            raise ValueError(
```

Comment:
Aren't these prime candidates for more a more specific exception type?

`InvalidColumnName` or somesuch? Or maybe use pandas's own `InvalidIndexError(Exception)` 🤷  ?

Reply:
These are meant as feedback to users, and the entire user input (a table in xlsx/csv) is regarded as "one" value in this respect, and thereby I have picked ValueError.